nuclear/roles/NUClearRole.cmake

```diff
@@ -42,6 +42,9 @@ function(NUCLEAR_ROLE)
   # modules are used in the role
   set_target_properties(${role} PROPERTIES NUCLEAR_MODULES "${role_modules}")
 
+  # Add warnings for the roles
+  target_compile_options(${role} PRIVATE ${NUCLEAR_ROLE_WARNINGS})
```

Comment:
Yeah this one here is where it becomes ambiguous with the variable, are you referring to NUCLEAR_ROLE as in this role, or NUCLEAR_ROLES as in the system?

Reply:
Amended :+1: